creator-node/src/fileManager.js

```diff
@@ -95,36 +96,32 @@ async function saveFileToIPFSFromFS (req, srcPath) {
  */
 async function saveFileForMultihash (req, multihash, expectedStoragePath, gatewaysToTry, fileNameForImage = null) {
   try {
-    const storagePath = req.app.get('storagePath') // should be `/file_storage'
-
     // will be modified to directory compatible route later if directory
     // TODO - don't concat url's by hand like this, use module like urljoin
     let gatewayUrlsMapped = gatewaysToTry.map(endpoint => `${endpoint.replace(/\/$/, '')}/ipfs/${multihash}`)
 
-    // Check if the file we are trying to copy is in a directory and if so, create the directory first
-    // E.g if the expectedStoragePath includes a dir like /file_storage/QmABC/Qm123, path.parse gives us
-    // { root: '/', dir: '/file_storage/QmABC', base: 'Qm123',ext: '', name: 'Qm2' }
-    // but if expectedStoragePath is a file like /file_storage/Qm123, path.parse gives us
-    // { root: '/', dir: '/file_storage', base: 'Qm123', ext: '', name: 'Qm123' }
-    // In the case where the parsed expectedStoragePath dir contains storagePath (`/file_storage`)
-    // but does not equal it, we know that it's a directory
     const parsedStoragePath = path.parse(expectedStoragePath).dir
-    if (parsedStoragePath !== storagePath && parsedStoragePath.includes(storagePath) && fileNameForImage) {
-      const splitPath = parsedStoragePath.split('/')
-      // parsedStoragePath looks like '/file_storage/Qm123' for a dir and when you call split, the result is
-      // ['', 'file_storage', 'Qm123']. the third item in the array is the directory cid, so index 2
-      if (splitPath.length !== 3) throw new Error(`Invalid expectedStoragePath for directory ${expectedStoragePath}`)
 
-      // override gateway urls to make it compatible with directory
-      gatewayUrlsMapped = gatewaysToTry.map(endpoint => `${endpoint.replace(/\/$/, '')}/ipfs/${splitPath[2]}/${fileNameForImage}`)
+    try {
+      // calling this on an existing directory doesn't overwrite the existing data or throw an error
+      // the mkdir recursive is equivalent to `mkdir -p`
+      await mkdir(parsedStoragePath, { recursive: true })
```

Comment:
why use this instead of ensureDir?

Reply:
no reason except ensureDir is not node fs standard lib so just used this function instead

Reply:
oh interesting should we not use it elsewhere then?

Reply:
cc @SidSethi

Reply:
not that we shouldn't use ensureDir, i just wanted to keep the functionality consistent with what was there before. we had recursive true so i just kepp that

Reply:
👍 should move over eventually but agreed doesn't need to be in this PR

Reply:
agreed, resolving this chain